testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/exportimport/ExportImportTest.java

```diff
@@ -262,6 +271,45 @@ public void testImportWithNullAuthenticatorConfigAndNoDefaultBrowserFlow() {
         Assert.assertTrue("Imported realm hasn't been found!", isRealmPresent("cez"));
     }
 
+    @Test
+    @EnableFeature(value = Profile.Feature.DECLARATIVE_USER_PROFILE)
```

Comment:
I had a look why this test fails, and I found that once you add the `@EnableFeature` annotation on a method, it will restart the Quarkus instance with the new configuration. 

https://github.com/keycloak/keycloak/blob/82269f789ab0a336ea33788216a94b2b08002744/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/KeycloakContainerFeaturesController.java#L147-L150

Once that happens, the referenced in `AbstractKeycloakTest` is now stale.

I found that the test works as expected without that annotation. 

I would be ok to remove this annotation, and open a follow-up issue to re-think the `adminClient` in `AbstractKeycloakTest`. Maybe it would be better to get rid of the attribute `adminClient` and to only keep the method `getAdminClient()` and make it retrieve the latest instance from `testContext` instead. There are a lot of references that need to be updated, therefore it should be a different issue.

Reply:
Thank you, your findings are inline with mine. I was experimenting with `skipRestart` flag: `@EnableFeature(value = Profile.Feature.DECLARATIVE_USER_PROFILE, skipRestart = true)` and it also resolves the test failure.

I was digging a little bit and found out that the reason why the test is passing without the annotation is the fact the rest endpoints in the `UserProfileResource` are not guarded by feature flag. I've discussed it with @pedroigor and it turned out there is no plan to add such a guard there since the User Profle will be supported soon and that "it does not expose anything critical/sensitive to when declarative is not used". Therefore I decided to follow your suggestion to remove the annotation.

I think it makes a lot of sense to open the issue to re-think the `adminClient` concept, maybe it might make sense to restrict the access to the field altogether and access it via a method only in whole testsuite. I'll create an issue for it.

Reply:
https://github.com/keycloak/keycloak/issues/24190